timeinterval/timeinterval.go

```diff
@@ -27,31 +27,56 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
-// Intervener determines whether a given time and active route time interval should mute outgoing notifications.
+// MuteIntervener determines whether the mute time interval should mute outgoing notifications.
 // It implements the TimeMuter interface.
-type Intervener struct {
+type MuteIntervener struct {
 	intervals map[string][]TimeInterval
 }
 
-func (i *Intervener) Mutes(names []string, now time.Time) (bool, error) {
+func (i *MuteIntervener) Mutes(names []string, now time.Time) (bool, error) {
 	for _, name := range names {
 		interval, ok := i.intervals[name]
 		if !ok {
-			return false, fmt.Errorf("time interval %s doesn't exist in config", name)
+			return false, fmt.Errorf("mute time interval %s doesn't exist in config", name)
 		}
-
 		for _, ti := range interval {
 			if ti.ContainsTime(now.UTC()) {
 				return true, nil
 			}
 		}
 	}
-
 	return false, nil
 }
 
-func NewIntervener(ti map[string][]TimeInterval) *Intervener {
-	return &Intervener{
+func NewMuteIntervener(ti map[string][]TimeInterval) *MuteIntervener {
+	return &MuteIntervener{
+		intervals: ti,
+	}
+}
+
+// ActiveIntervener determines whether the active time interval should mute outgoing notifications.
+// It implements the TimeMuter interface.
+type ActiveIntervener struct {
+	intervals map[string][]TimeInterval
+}
+
+func (i *ActiveIntervener) Mutes(names []string, now time.Time) (bool, error) {
```

Comment:
I'm not a fan of this - this is pretty much the same logic as the other but with a flipped conditional. I'd rather keep it as is and continue to have the `!muted` on the stage itself.

This seems like a lot of code to change/add for the sake of a flipping a conditional.

Reply:
That's OK. We'll then need to put the marker back into `TimeMuteStage` and `TimeActiveStage` though as otherwise `Intervener` doesn't know if it's activing or muting.

Reply:
Maybe we should introduce it then? It's hard for me to see the benefit of this as is. 😭

Reply:
I do think making time active intervals adhere to the contract of `Mute` is a good win in of itself. Mute no longer returns false when it should be muted, and true when it shouldn't be muted.

Reply:
That still doesn't work I'm afraid, because `Intervener` doesn't know if the intervals in `names` are active time intervals or mute time intervals.